### ISSUE TITLE
Error 'No gradients provided for any variable' occurs

### ISSUE THREAD

**Issue by YuanhuiCheng** [Mazen-Abdelbadea/SYSML-Network-Requirements#1]
Hi, I'm writing a simple Tensorflow app that involves a coordinator, a ps, and multiple workers. I borrowed your code from [main_ps_resnet_cifar10.py](https://github.com/Mazen-Abdelbadea/SYSML-Network-Requirements/blob/master/Tensorflow/Python/main_ps_resnet_cifar10.py) and tried to make it work. However, the following error occurs when running this piece of code `optimizer.apply_gradients(zip(gradients, model.trainable_variables))`,
```log
No gradients provided for any variable: ['conv2d/kernel:0', 'conv2d/bias:0', 'conv2d_1/kernel:0', 'conv2d_1/bias:0', 'conv2d_2/kernel:0', 'conv2d_2/bias:0', 'dense/kernel:0', 'dense/bias:0', 'dense_1/kernel:0', 'dense_1/bias:0'].
```
So far I cannot figure it work, could u tell me that if u met the same error when u run your code? If so, could u help me solve the problem? thanks.

**Comment by Mazen-Abdelbadea** [comment]
Hi, 

I believe I received similar error when trying 3-CNN model in parameter server setup but I didn't attempt to solve it since I wanted to run Resnet50 model. 
When I tried Resnet50 model, I received another error `Sharded variable doesn’t support being used as a reference.` which I couldn't solve for 2 weeks. I switched to Pytorch after struggling with TensorFlow to run ParameterServer. 
If you have new updates or could run Parameter Server eventually, please let me know.
Thanks.
